pkg/v1/sdk/capabilities/discovery/cluster_gvr.go

```diff
@@ -12,19 +13,41 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// Group represents any API group that may exist on a cluster, with ability to specify versions and resource to check for GVR.
+var (
+	// ErrInvalidWithResourceMethodArgument occurs when an empty string argument is passed into the WithResource method.
+	ErrInvalidWithResourceMethodArgument = errors.New("WithResource method must have non-empty string argument; omit method to indicate any resource")
```

Comment:
If we make these errors private, users of our API will no longer be able to use the standard library method `errors.Is(...)` to determine if a certain error happened. They would be forced to match on error string using something like `strings.Contains(...)`, which I think is not as clean and has more room for mistakes.

I am thinking maybe we can have the errors public/exported like the  io package does: https://pkg.go.dev/io#pkg-variables

On the other hand, I'm not sure if users of our API will ever want to be able to match on the the  `ErrInvalidWithResourceMethodArgument` or `ErrInvalidWithVersionsMethodArgument` errors. If we can't see them ever wanting to be able to look for a particular error, then I agree that I should make these errors private. What do you think?

I'm thinking even if our API users were to write tests for themselves, they might want to look out for these two invalid method arguments errors.

Reply:
> On the other hand, I'm not sure if users of our API will ever want to be able to match on the the ErrInvalidWithResourceMethodArgument or ErrInvalidWithVersionsMethodArgument errors.

This is really the intention behind my earlier comment. I don't see these errors being too useful for consumers. I also don't think anyone writing unit tests will be testing validation behavior of our underlying APIs (or they shouldn't at least). A good proxy for consumer usage is to look at `pkg/v1/sdk/capabilities/discovery/tkg` and see how the tests are written.

I don't feel there is a strong enough reason, so I'm inclined to avoid another API contract with exported errors that makes it harder to change later.

Reply:
> I don't see these errors being too useful for consumers. I also don't think anyone writing unit tests will be testing validation behavior of our underlying APIs (or they shouldn't at least)...I don't feel there is a strong enough reason, so I'm inclined to avoid another API contract 

Yup, if consumers have no use for these errors, I should make them private.

Errors are now not exported.